src/compas/geometry/xforms/quaternion.py

```diff
@@ -0,0 +1,225 @@
+
+
+from compas.geometry.transformations import quaternion_multiply
+from compas.geometry.transformations import quaternion_conjugate
+from compas.geometry.transformations import quaternion_unitize
+from compas.geometry.transformations import quaternion_canonize
+from compas.geometry.transformations import quaternion_norm
+from compas.geometry.transformations import quaternion_is_unit
+
+from compas.geometry.basic import allclose
+
+__all__ = ['Quaternion']
+
+
+class Quaternion(object):
+
+    """Creates a ``Quaternion`` object.
+
+    Parameters
+    ----------
+    w : float
+        The scalar (real) part of a quaternion.
+    x, y, z : float
+        Components of the vector (complex, imaginary) part of a quaternion.
+
+
+    Examples
+    --------
+    >>> Q = Quaternion(1.0, 1.0, 1.0, 1.0).unitized()
+    >>> R = Quaternion(0.0,-0.1, 0.2,-0.3).unitized()
+    >>> P = R*Q
+    >>> P.is_unit
+    True
+
+    Notes
+    -----
+    The default convention to represent a quaternion :math:`q` in this module is by four real values **w**, **x**, **y**, **z**.
+    The first value **w** is the scalar (real) part, and **x**, **y**, **z** form the vector (complex, imaginary) part [1]_, so that:
+
+    :math:`q = w + xi + yj + zk`
+
+    where :math:`i, j, k` are basis components with following multiplication rules [2]_:
+
+    :math:`ii = jj = kk = ijk = -1`
```

Comment:
`:math:` is typically used for inline math. to add a block of math stuff use `.. math::`